src/sprig-matrix.ts

```diff
@@ -0,0 +1,553 @@
+    return GL.transformMat4(out ?? tmp(), v1, v2) as T;
+  }
+
+  export function zero(out?: T): T {
+    return GL.zero(out ?? tmp()) as T;
+  }
+}
+
+export module quat {
+  export type T = quat;
+  export type InputT = T | readonly [number, number, number, number];
+  const GL = GLM.quat;
+
+  export function tmp(): T {
+    return tmpArray(4);
+  }
+
+  export function create(): T {
+    return float32ArrayOfLength(4);
+  }
+
+  export function clone(v: InputT): T {
+    return GL.clone(v) as T;
+  }
+
+  export function copy(out: T, v1: InputT): T {
+    return GL.copy(out, v1) as T;
+  }
+
+  export const IDENTITY = identity(create());
+
+  export function equals(v1: InputT, v2: InputT): boolean {
+    return GL.equals(v1, v2);
+  }
+  export function exactEquals(v1: InputT, v2: InputT): boolean {
+    return GL.exactEquals(v1, v2);
+  }
+
+  export function add(v1: InputT, v2: InputT, out?: T): T {
+    return GL.add(out ?? tmp(), v1, v2) as T;
+  }
+  export function mul(v1: InputT, v2: InputT, out?: T): T {
+    return GL.mul(out ?? tmp(), v1, v2) as T;
+  }
+  export function slerp(v1: InputT, v2: InputT, n: number, out?: T): T {
+    return GL.slerp(out ?? tmp(), v1, v2, n) as T;
+  }
+
+  export function normalize(v1: InputT, out?: T): T {
+    return GL.normalize(out ?? tmp(), v1) as T;
+  }
+  export function identity(out?: T): T {
+    return GL.identity(out ?? tmp()) as T;
+  }
+
+  export function conjugate(v1: InputT, out?: T): T {
+    return GL.conjugate(out ?? tmp(), v1) as T;
+  }
+  export function invert(v1: InputT, out?: T): T {
+    return GL.invert(out ?? tmp(), v1) as T;
+  }
+
+  export function setAxisAngle(axis: vec3.InputT, rad: number, out?: T): T {
+    return GL.setAxisAngle(out ?? tmp(), axis, rad) as T;
+  }
+  export function getAxisAngle(q: InputT, out?: vec3.T): number {
+    return GL.getAxisAngle(out ?? tmp(), q);
+  }
+  export function getAngle(q1: InputT, q2: InputT): number {
+    return GL.getAngle(q1, q2);
+  }
+
+  export function rotateX(v1: InputT, n: number, out?: T) {
+    return GL.rotateX(out ?? tmp(), v1, n) as T;
+  }
+  export function rotateY(v1: InputT, n: number, out?: T) {
+    return GL.rotateY(out ?? tmp(), v1, n) as T;
+  }
+  export function rotateZ(v1: InputT, n: number, out?: T) {
+    return GL.rotateZ(out ?? tmp(), v1, n) as T;
+  }
+  export function fromEuler(x: number, y: number, z: number, out?: T): T {
+    return GL.fromEuler(out ?? tmp(), x, y, z) as T;
+  }
+}
+
+export module mat4 {
+  export type T = mat4;
+  // prettier-ignore
+  export type InputT = T | readonly [number, number, number, number,
+                                     number, number, number, number,
+                                     number, number, number, number,
+                                     number, number, number, number];
+  const GL = GLM.mat4;
+
+  export function tmp(): T {
+    return tmpArray(16);
+  }
+
+  export function create(): T {
+    return float32ArrayOfLength(16);
```

Comment:
Very important bug here!
mat4.create() creates an identity matrix, not a blank matrix
same w/ mat3, quat, etc